nipc-openapi/NIPC.yaml

```diff
@@ -0,0 +1,1622 @@
+                  - $ref: '#/components/schemas/SdfObject'
+                  - $ref: '#/components/schemas/Id'
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorized
+        '405':
+          description: Invalid request
+        '500':
+          description: Server-side failure
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/FailureResponse' 
+                  - $ref: '#/components/schemas/RegistrationErrorCodes'
+ 
+    delete:
+      tags:
+        - NIPC registration APIs
+      summary: Delete an sdfObject
+      description: |-
+        Delete an sdfObject, including Properties, Events and actions
+      operationId: deleteSdfObject
+      parameters:
+        - name: id
+          in: path
+          description: device or group id the model is to be registered for
+          required: true
+          schema:
+            type: string
+            format: uuid
+            example: 12345678-1234-5678-1234-56789abcdef4
+        - name: sdfobject
+          in: path
+          description: sdfObject name
+          required: true
+          schema:
+            type: string
+            example: "https://example.com/heartrate/sdfObject/healthsensor"
+      responses:
+        '200':
+          description: Success
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/SuccessResponse'
+                  - $ref: '#/components/schemas/SdfObject'
+                  - $ref: '#/components/schemas/Id'
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorized
+        '405':
+          description: Invalid request
+        '500':
+          description: Server-side failure
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/FailureResponse' 
+                  - $ref: '#/components/schemas/RegistrationErrorCodes'
+
+    get:
+      tags:
+        - NIPC registration APIs
+      summary: get a registered sdfObject
+      description: |-
+        Get an sdfObject, including Properties, Events and actions
+      operationId: getSdfObject
+      parameters:
+        - name: id
+          in: path
+          description: device or group id the model is to be registered for
+          required: true
+          schema:
+            type: string
+            format: uuid
+            example: 12345678-1234-5678-1234-56789abcdef4
+        - name: sdfobject
+          in: path
+          description: sdfObject name
+          required: true
+          schema:
+            type: string
+            example: "https://example.com/heartrate/sdfObject/healthsensor"
+      responses:
+        '200':
+          description: Success
+          content:
+            application/json:
```

Comment:
I believe this endpoint should return a valid `application/sdf+json` object. Suggestion:

- Omit `requestID` in the response body and add it to the header.
- Omit `status` in the response body as it can be deduced by the response code, e.g., 200 = SUCCESS.
- Change `content` to `application/sdf+json` ✅

Reply:
Thanks Lorenzo, I made the suggested updates